### DISCUSSION TITLE
CS tools install failed with Python error

### DISCUSSION THREAD

**Discussion by KrishTSGitHubUser** [thoughtspot/cs_tools#100] [General]
python _bootstrapper.py --install failing with error "No Python at '"**C:\Program Files\Python312\python.exe'" My current python version is Python 3.11.6 and I don't have  any directory C:\Program Files\Python312 in windows.

**Comment by boonhapus** [comment]
@KrishTSGitHubUser can you share the output of `python -m sysconfig` ?

**Comment by KrishTSGitHubUser** [comment]
Herr is the dump of below command.

python -m sysconfig > x.txt

Platform: "win-amd64"
Python version: "3.11"
Current installation scheme: "nt"

Paths:
                data = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311"
                include = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311\Include"
                platinclude = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311\Include"
                platlib = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311\Lib\site-packages"
                platstdlib = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311\Lib"
                purelib = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311\Lib\site-packages"
                scripts = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311\Scripts"
                stdlib = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311\Lib"

Variables:
                BINDIR = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311"
                BINLIBDEST = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311\Lib"
                EXE = ".exe"
                EXT_SUFFIX = ".cp311-win_amd64.pyd"
                INCLUDEPY = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311\Include"
                LIBDEST = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311\Lib"
                TZPATH = ""
                VERSION = "311"
                VPATH = "..\.."
                abiflags = ""
                base = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311"
                exec_prefix = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311"
                installed_base = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311"
                installed_platbase = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311"
                platbase = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311"
                platlibdir = "DLLs"
                prefix = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311"
                projectbase = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311"
                py_version = "3.11.6"
                py_version_nodot = "311"
                py_version_nodot_plat = "311"
                py_version_short = "3.11"
                srcdir = "C:\Users\krish.srinivasan\AppData\Local\Programs\Python\Python311"
                userbase = "C:\Users\krish.srinivasan\AppData\Roaming\Python"

Thanks
Krish

**Reply by boonhapus** [reply]
I see, thank you @KrishTSGitHubUser . Can you check this path to see if there's a directory here? If so, please delete it and then re-run the original bootstrapper command.

`C:\Users\krish.srinivasan\AppData\Roaming\cs_tools\.cs_tools`

**Comment by KrishTSGitHubUser** [comment]
Thank you for help. Python 312 error is resolved. Now I got different errors.  Please see the attached error log cs_tools-bootstrap-error-tmp9dnpjf4t.log.

Thanks
Krish

**Reply by boonhapus** [reply]
@KrishTSGitHubUser I think you'll need to attach it to the Discussion in GH.

https://github.com/thoughtspot/cs_tools/discussions/100#discussioncomment-8259848

Email response won't attach the file.

**Comment by KrishTSGitHubUser** [comment]
[cs_tools-bootstrap-error-tmp9dnpjf4t.log](https://github.com/thoughtspot/cs_tools/files/14068026/cs_tools-bootstrap-error-tmp9dnpjf4t.log)
See error attached...

**Reply by boonhapus** [reply]
@KrishTSGitHubUser it appears your network security team does not allow installing from `pypi`. Do you know if you have a proxy server you can route traffic through?

```
Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1006)'))': /packages/15/aa/3f4c7bcee2057a76562a5b33ecbd199be08cdb4443a02e26bd2c3cf6fc39/pip-23.3.2-py3-none-any.whl.metadata
```
